src/demo/natalyj-demo/SplitColorShaderMaterial/useSplitColorShaderMaterial.ts

```diff
@@ -17,58 +18,63 @@ import Fox from '../models/Fox/Fox.gltf';
 import { CameraController } from '../../../CameraController';
 import { createRenderer, resizeRenderer } from '../../../util';
 
-import {
-  ColorConfig,
-  SplitColorShaderMaterial,
-} from './SplitColorShaderMaterial';
+import { SplitColorShaderMaterial } from './SplitColorShaderMaterial';
 import { createCustomGUI } from './createCustomGUI';
+import { ColorConfig, AvailableObjects, GUIConfig } from './types';
+
+const FAR = 400;
+const RADIUS = 200;
+const BOX_SIZE = 100;
+const HELPER_SIZE = 300;
 
 export function useSplitColorShaderMaterial(): Demo {
   const renderer = createRenderer();
-  const camera = new PerspectiveCamera(75, 1, 0.01, 0.8);
-  const cameraController = new CameraController(0.2, 0.01);
+  const camera = new PerspectiveCamera(75, 1, 0.01, FAR);
+  const cameraController = new CameraController(RADIUS, 0.01);
   const scene = new Scene();
 
   const colorConfig: ColorConfig = {
+    negativeColor: new Color('green'),
+    positiveColor: new Color('red'),
     planeNormal: new Vector3(0.7, 0.3, 0.1),
-    distanceFromOrigin: -0.025,
+    distanceFromOrigin: 2,
   };
 
-  const material = new SplitColorShaderMaterial(
-    new Color('green'),
-    new Color('red'),
-    colorConfig
-  );
+  const material = new SplitColorShaderMaterial(colorConfig);
+
+  const geometry = new BoxGeometry(BOX_SIZE, BOX_SIZE, BOX_SIZE);
+  const box = new Mesh(geometry, material);
+  scene.add(box);
+
+  const possibleObjects = new Map<AvailableObjects, Object3D>();
+  possibleObjects.set(AvailableObjects.Box, box);
 
   const loader = new GLTFLoader();
-  loader.load(
-    Fox,
-    (gltf) => {
-      gltf.scene.traverse((obj) => {
-        if (obj instanceof Mesh) {
-          obj.material = material;
-          obj.geometry.scale(0.001, 0.001, 0.001);
-        }
-      });
-      scene.add(gltf.scene);
-    },
-    () => {},
-    () => {
-      const geometry = new BoxGeometry(0.1, 0.1, 0.1);
-      const box = new Mesh(geometry, material);
-      scene.add(box);
-    }
-  );
+  loader.load(Fox, (gltf) => {
+    gltf.scene.traverse((obj) => {
+      if (obj instanceof Mesh) {
+        obj.material = material;
+      }
+    });
+    possibleObjects.set(AvailableObjects.Fox, gltf.scene);
+    gltf.scene.visible = false;
```

Comment:
Please pay attention: I keep two models in a scene and switch between their visibility values (fo a selector control). Another option is to add/remove objects from a scene. I'm not sure what way is better.